src/coap.c

```diff
@@ -492,7 +490,7 @@ static uint32_t coap_calculate_packet_size(struct ns_coap_message *cm,
 /*
  * Composes CoAP message from ns_coap_message structure.
  * Returns 0 on success or bit mask with wrong field.
- * Packet returned must be explicitly deleted by NS_FREE.
+ * Packet returned must be explicitly freed by iobuf_free.
```

Comment:
Well not necessarily. The whole point of iobuffers is that they can be reused. I'd just drop this line; the behaviour is aligned with the fossa buffer management patterns.